templates/swift/Sources/Services/Realtime.swift.twig

```diff
@@ -38,17 +37,30 @@ open class Realtime : Service {
 
         if (socketClient != nil) {
             reconnect = false
-            closeSocket()
-        } else {
-            socketClient = WebSocketClient(url, tlsEnabled: !client.selfSigned, delegate: self)!
+            try await closeSocket()
         }
 
-        try! socketClient?.connect()
+        socketClient = WebSocketClient(
+            url,
+            tlsEnabled: !client.selfSigned,
+            delegate: self
+        )
+
+        try await socketClient?.connect()
```

Comment:
I'm having trouble testing this. I updated the playground for apple's subscribe call [here](https://github.com/appwrite/playground-for-apple-swiftui/blob/50c601345421346dfd6fe330e201537a929cc96e/PlaygroundForApple/PlaygroundViewModel.swift#L138-L144) to be:

```swift
    func subscribe() async {
        do {
            try await realtime.subscribe(channel: "databases.\(databaseId).collections.\(collectionId).documents") { event in
                DispatchQueue.main.async {
                    self.message = String(describing: event.payload!)
                }
            }
    
            try await realtime.subscribe(channel: "files") { event in
                DispatchQueue.main.async {
                    self.message = String(describing: event.payload!)
                }
            }
        } catch {
            dialogText = error.localizedDescription
        }
    }
```

but it seems like the files event isn't coming through. The document one does, though. And if I swap them like:

```swift
    func subscribe() async {
        do {
            try await realtime.subscribe(channel: "files") { event in
                DispatchQueue.main.async {
                    self.message = String(describing: event.payload!)
                }
            }

            try await realtime.subscribe(channel: "databases.\(databaseId).collections.\(collectionId).documents") { event in
                DispatchQueue.main.async {
                    self.message = String(describing: event.payload!)
                }
            }
        } catch {
            dialogText = error.localizedDescription
        }
    }
```

The files one will come through, but the documents one won't. 

Any ideas on what I'm doing wrong?

Reply:
Hmm in my tests I had a delay between subscribes, might be an issue with the debounce. Will check

Reply:
Fixed